src/machine_charm.py

```diff
@@ -26,13 +26,16 @@
 class MachineSubordinateRouterCharm(abstract_charm.MySQLRouterCharm):
     """MySQL Router machine subordinate charm"""
 
+    READ_WRITE_PORT = 6446
+    READ_ONLY_PORT = 6447
+
```

Comment:
Should they be in constants as nowadays available for both VM and K8s?

Reply:
moved constants to the abstract charm file in ee4b09862143abb00c36232590ac72f383d91e5b